### PR TITLE
Quick search projects

### PR DESCRIPTION
**Pull request by ayushgml**
This PR closes the following issue:
 - #3983 

## Changes

- Added a Quick Project Search filter in All Projects tab. Logic:
```
if (typeFilter === 'all') {
      return searchInput.trim() === '' ? projects : projects.filter(p => p.name?.toLowerCase().includes(searchInput));
    }

    if (typeFilter === 'git') {
      return searchInput.trim() === ''
        ? projects.filter(p => p.isGitRepo)
        : projects.filter(p => p.name?.toLowerCase().includes(searchInput) && p.isGitRepo);
    }
    return searchInput.trim() === ''
      ? projects.filter(p => !p.isGitRepo)
      : projects.filter(p => p.name?.toLowerCase().includes(searchInput) && !p.isGitRepo);
```

- Highlighting of the searched query in the project name. Logic:
```
  const highlightQuery = (text: string) => {
    if (!query?.trim()) {
      return <S.Span>{text}</S.Span>;
    }
    const parts = text.split(new RegExp(`(${query})`, 'gi'));
    return (
      <S.Span>
        {parts.map(part => (part.toLowerCase() === query.toLowerCase() ? <S.BoldSpan>{part}</S.BoldSpan> : part))}
      </S.Span>
    );
  };
```

## How to test it

- Run the desktop app and go to "All Projects" tab

## Screenshots
Screenshot 2023-05-26 at 2 29 32 PM

Screenshot 2023-05-26 at 2 30 16 PM

Screenshot 2023-05-26 at 2 30 47 PM

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/components/molecules/ProjectsList/ProjectsList.tsx [hunk 1]
```diff
@@ -55,17 +56,20 @@ const ProjectsList: React.FC<IProps> = props => {
     );
 
     if (typeFilter === 'all') {
-      return projects;
+      return searchInput.trim() === '' ? projects : projects.filter(p => p.name?.toLowerCase().includes(searchInput));
```
Comment (devcatalin) [inline src/components/molecules/ProjectsList/ProjectsList.tsx:59]:
```suggestion
      return searchInput.trim() === '' ? projects : projects.filter(p => p.name?.toLowerCase().includes(searchInput.toLowerCase()));
```

File: src/components/molecules/ProjectsList/ProjectsList.tsx [hunk 2]
```diff
@@ -55,17 +56,20 @@ const ProjectsList: React.FC<IProps> = props => {
...
     }
 
     if (typeFilter === 'git') {
-      return projects.filter(p => p.isGitRepo);
+      return searchInput.trim() === ''
+        ? projects.filter(p => p.isGitRepo)
+        : projects.filter(p => p.name?.toLowerCase().includes(searchInput) && p.isGitRepo);
```
Comment (devcatalin) [inline src/components/molecules/ProjectsList/ProjectsList.tsx:65]:
```suggestion
        : projects.filter(p => p.name?.toLowerCase().includes(searchInput.toLowerCase()) && p.isGitRepo);
```

File: src/components/molecules/ProjectsList/ProjectsList.tsx [hunk 3]
```diff
@@ -55,17 +56,20 @@ const ProjectsList: React.FC<IProps> = props => {
...
     }
+    return searchInput.trim() === ''
+      ? projects.filter(p => !p.isGitRepo)
+      : projects.filter(p => p.name?.toLowerCase().includes(searchInput) && !p.isGitRepo);
```
Comment (devcatalin) [inline src/components/molecules/ProjectsList/ProjectsList.tsx:69]:
```suggestion
      : projects.filter(p => p.name?.toLowerCase().includes(searchInput.toLowerCase()) && !p.isGitRepo);
```

**Review by devcatalin — CHANGES_REQUESTED** [review]
Thanks for your contribution, left some quick suggestions!

**Comment by ayushgml** [comment]
@devcatalin I have made the changes.

**Review by devcatalin — APPROVED** [review]

**Comment by devcatalin** [comment]
Great, thank you, merging this!

**Pull request merged**